recipes/resource_pool/all/test_package/coro.cc

```diff
@@ -0,0 +1,41 @@
+#include <yamail/resource_pool/async/pool.hpp>
+
+#include <atomic>
+#include <iostream>
+#include <thread>
+#include <memory>
+
+using ofstream_pool = yamail::resource_pool::async::pool<std::unique_ptr<std::ostream>>;
+using time_traits = yamail::resource_pool::time_traits;
+
+struct null_buffer : std::streambuf { int overflow(int c) override { return c; } } null_buf;
+
+int main() {
+    std::atomic_bool success{true};
+    boost::asio::io_context service;
+    ofstream_pool pool(1, 10);
+    boost::asio::spawn(service, [&](boost::asio::yield_context yield){
+        boost::system::error_code ec;
+        auto handle = pool.get_auto_waste(service, yield[ec], time_traits::duration::max());
+        if (ec) {
+            std::cout << "handle error: " << ec.message() << std::endl;
+            success = false;
+            return;
+        }
+        std::cout << "got resource handle" << std::endl;
+        if (handle.empty()) {
+            auto stream = std::make_unique<std::ostream>(&null_buf);
+            if (!stream->good()) {
+                success = false;
+                return;
+            }
+            handle.reset(std::move(stream));
+        }
+        *(handle.get()) << (time_traits::time_point::min() - time_traits::now()).count() << std::endl;
+        if (handle.get()->good()) {
+            handle.recycle();
+        }
+    });
+    service.run();
+    return success ? 0 : 1;
```

Comment:
We like to test the packaging and not the underlying library

```suggestion
    return 0;
```